lib/backend/k8s/conversion/conversion_test.go

```diff
@@ -215,19 +219,20 @@ var _ = Describe("Test Pod conversion", func() {
 		Expect(err).NotTo(HaveOccurred())
 
 		// Assert key fields.
-		Expect(wep.Key.(model.ResourceKey).Name).To(Equal("podA"))
+		Expect(wep.Key.(model.ResourceKey).Name).To(Equal("nodeA-k8s-podA-eth0"))
+		Expect(wep.Key.(model.ResourceKey).Namespace).To(Equal("default"))
 		Expect(wep.Key.(model.ResourceKey).Kind).To(Equal(apiv2.KindWorkloadEndpoint))
 		// Assert value fields.
-		Expect(wep.Value.(*apiv2.WorkloadEndpoint).Spec.Workload).To(Equal("default.podA"))
+		//Expect(wep.Value.(*apiv2.WorkloadEndpoint).Spec.Workload).To(Equal("default.podA"))
+		Expect(wep.Value.(*apiv2.WorkloadEndpoint).Spec.Workload).To(Equal("nodeA-k8s-podA-eth0"))
 		Expect(wep.Value.(*apiv2.WorkloadEndpoint).Spec.Node).To(Equal("nodeA"))
 		Expect(wep.Value.(*apiv2.WorkloadEndpoint).Spec.Endpoint).To(Equal("eth0"))
 		Expect(wep.Value.(*apiv2.WorkloadEndpoint).Spec.Orchestrator).To(Equal("k8s"))
 		Expect(len(wep.Value.(*apiv2.WorkloadEndpoint).Spec.IPNetworks)).To(Equal(1))
 		Expect(wep.Value.(*apiv2.WorkloadEndpoint).ObjectMeta.Labels).To(Equal(map[string]string{"calico/k8s_ns": "default"}))
-		//Expect(wep.Value.(*apiv2.WorkloadEndpoint).State).To(Equal("active"))
 	})
 
-	It("should Parse a Pod with no NodeName", func() {
+	It("should not parse a Pod with no NodeName", func() {
```

Comment:
Hm, this is worrying.

I think it's probably fine, but I don't know off the top of my head which logic relies on this test passing as it is today. This could very well break something down the line.

Reply:
I jumped the gun with "fixing" this test. At a minimum I should add a note about this restriction in the appropriate conversion method.
@caseydavenport Will our different clients be interested in knowing about a `Pod` that isn't assigned to a `Node` yet?

 @robbrockbank The `WorkloadEndpoint` name goes something like `<node>-k8s-<pod>-<endpoint>`. Looks like there may be cases when a Pod exists without a `Node`. Do you have any thoughts on this?

Reply:
Previously the WEP had the node as an explicit index, so in a way nothing has really changed.